sklearn/pipeline.py

```diff
@@ -20,12 +21,74 @@
 from .externals import six
 from .utils import tosequence
 from .utils.metaestimators import if_delegate_has_method
-from .externals.six import iteritems
 
 __all__ = ['Pipeline', 'FeatureUnion']
 
 
-class Pipeline(BaseEstimator):
+class _BasePipeline(BaseEstimator):
```

Comment:
In python3 `__metaclass__` is no more, else where `six.with_metaclass` is used. Change to:

`class _BasePipeline(six.with_metaclass(ABCMeta, BaseEstimator)):`

and remove L31